spot_rl_experiments/spot_rl/utils/robot_subscriber.py

```diff
@@ -81,18 +86,36 @@ def __init__(self, spot=None, *args, **kwargs):
 
         rospy.loginfo(f"[{self.node_name}]: Robot subscribing has started.")
 
+    # Letting the user know about timeout to get images. This function lets the user know time taken to get the topics
+    def wait_for_images(self, subscriptions, timeout=30):
+        start_time = rospy.get_time()
+        while not all([self.msgs[s] is not None for s in subscriptions]):
+            if rospy.get_time() - start_time > timeout:
+                rospy.logwarn(f"[{self.node_name}]: Timeout waiting for image topics.")
+                break
+            rospy.sleep(1)
+
+    # Added Try-catch statement to debug errors from Image callbacks
     def img_callback(self, topic, msg):
-        self.msgs[topic] = msg
-        self.updated[topic] = True
+        try:
+            self.msgs[topic] = msg
+            self.updated[topic] = True
+        except Exception as e:
+            rospy.logerr(
+                f"[{self.node_name}]: Error processing message from topic {topic}: {e}"
+            )
 
+    # Added Try-catch statement to debug errors from processing robot state
     def robot_state_callback(self, msg):
-        x, y, yaw = msg.data[:3]
-        self.x, self.y, self.yaw = self.spot.xy_yaw_global_to_home(x, y, yaw)
-        self.current_arm_pose = msg.data[3:-7]
-        self.link_wr1_position, self.link_wr1_rotation = (
-            msg.data[-7:][:3],
-            msg.data[-7:][3:],
-        )
-
-    def msg_to_cv2(self, *args, **kwargs) -> np.array:
-        return self.cv_bridge.imgmsg_to_cv2(*args, **kwargs)
+        try:
```

Comment:
why does robot_state need to be in a try except?

Reply:
Added logging to know if robot_state ever fails.